test/testdata/durationcheck.go

```diff
@@ -1,9 +1,38 @@
 //args: -Edurationcheck
 package testdata
 
-import "time"
+import (
+	"fmt"
+	"time"
+)
 
-func waitFor(someDuration time.Duration) {
-	timeToWait := someDuration * time.Second // ERROR "Multiplication of durations: `someDuration \\* time.Second` "
+type durationCheckData struct {
+	i int
+	d time.Duration
+}
+
+func durationcheckCase01() {
+	dcd := durationCheckData{i: 10}
+	_ = time.Duration(dcd.i) * time.Second
+}
+
+func durationcheckCase02() {
+	dcd := durationCheckData{d: 10 * time.Second}
+	_ = dcd.d * time.Second // ERROR "Multiplication of durations: `dcd.d \\* time.Second`"
+}
+
+func durationcheckCase03() {
+	seconds := 10
+	fmt.Print(time.Duration(seconds) * time.Second)
```

Comment:
Why doesn't this yield an error? 🤔

Reply:
because `time.Duration` is a `int64` (`type Duration int64`) so in this case: `seconds` (`int`) is just converted to `int64` (`Duration`).

```
time.Duration(seconds) * time.Second
=> 10 * time.Second
```